python/lsst/ts/dream/csc/dream_csc.py

```diff
@@ -132,43 +178,129 @@ async def begin_disable(self, id_data: salobj.BaseDdsDataType) -> None:
         id_data: `CommandIdData`
             Command ID and data
         """
+        await self.disconnect()
         await self.cmd_disable.ack_in_progress(id_data, timeout=60)
         await super().begin_disable(id_data)
 
     async def disconnect(self) -> None:
         """Disconnect the DREAM CSC, if connected."""
         self.log.info("Disconnecting")
-        await self.model.disconnect()
+        self.weather_sleep_task.cancel()
+        await self.weather_loop_task
+        if self.model is not None:
+            await self.model.disconnect()
+        self.model = None
+        if self.mock:
+            await self.mock.close()
+            self.mock = None
+        self.log.info("Disconnected.")
 
     async def handle_summary_state(self) -> None:
         """Override of the handle_summary_state function to connect or
         disconnect to the DREAM CSC (or the mock client) when needed.
         """
-        self.log.info(f"handle_summary_state {salobj.State(self.summary_state).name}")
-        if self.disabled_or_enabled:
-            if not self.connected:
-                await self.connect()
-        else:
-            await self.disconnect()
+        self.log.info(
+            f"handle_summary_state {salobj.State(self.summary_state).name} [{self.connected=}]"
+        )
 
     async def configure(self, config: SimpleNamespace) -> None:
         self.config = config
 
+    async def connect_ess_remote(self) -> None:
+        """Establishes a connection to the weather station if not connected."""
+        if self.ess_remote is None:
+            self.ess_remote = salobj.Remote(domain=self.domain, name="ESS", index=301)
+            await self.ess_remote.start_task
+            self.log.info("Connected to ESS:301.")
+
+            self.weather_ok_flag = None
+            self.last_weather_update = 0.0
+
     @property
     def connected(self) -> bool:
-        return not self.model.connected
+        if self.model is None:
+            return False
+        return self.model.connected
 
     @staticmethod
     def get_config_pkg() -> str:
         return "ts_config_ocs"
 
     async def do_pause(self, data: salobj.BaseMsgType) -> None:
-        # To be implemented later.
-        pass
+        await self.disconnect()
 
     async def do_resume(self, data: salobj.BaseMsgType) -> None:
-        # To be implemented later.
-        pass
+        if not self.connected:
+            await self.connect()
+
+    async def weather_loop(self) -> None:
+        """Periodically checks weather station, and sends a flag if needed.
+
+        A weather flag is sent when the weather has changed, or every ten
+        minutes.
+        """
+        while self.model is not None and self.model.connected:
+            if self.simulation_mode == 0:
+                await self.connect_ess_remote()  # Connect to ESS if not already connected
```

Comment:
I am not happy with the way you use the ess_remove here and will get back to this after you have reviewed my comments.